doc/api/errors.md

```diff
@@ -1004,6 +1004,22 @@ Used when a callback is called more then once.
 can either be fulfilled or rejected but not both at the same time. The latter
 would be possible by calling a callback more then once.
 
+<a id="ERR_NET_INVALID_LISTEN_ARG"></a>
+### ERR_NET_INVALID_LISTEN_ARG
```

Comment:
@Trott @refack I am having trouble using the 'ERR_INVALID_ARG_TYPE' for the
`Server.prototype.listen()` in the `lib/net.js`. 
I am describing the problem below :

The `Server.prototype.listen()` implementation 
has some early return statements for the following cases:

`options = options._handle || options.handle || options;`
- `options instanceof TCP`
- `typeof options.fd === 'number' && options.fd >= 0`
- `(args.length === 0 || typeof args[0] === 'function' || (typeof options.port === 'undefined' && 'port' in options)) || typeof options.port === 'number' || typeof options.port === 'string'`
- `options.path && isPipeName(options.path)`

At the end, if none of these early return statements runs, then the error is thrown.

So I am having trouble using the 
`E('ERR_INVALID_ARG_TYPE', invalidArgType)` error. 
As I cannot compute a suitable error message using the 
`invalidArgType(name, expected, actual)` (file: `lib/internal/errors.js`) callback,
for the case that I described above.

I have not yet found a similar case in the source code.
That's why I used the new error `ERR_NET_INVALID_LISTEN_ARG` in the first place.
Do you have any suggestions on how should I handle this case or on how  should I use the `ERR_INVALID_ARG_TYPE` error for this case?

Should I refactor the `invalidArgType` method in the `lib/internal/errors.js`
to return a more generic error message (like: `invalid type`) 
if the `expected` argument is not specified?